chore(sdk-core): Refactor & split getVerifyAndSignBitGoChallenges

mohammadalfaiyazbitgo opened the pull request:
- Add method to sign bitgo challenge without verification. To be used in contexts where the caller will verify challenges/proofs separately from signing.

<!--
# Please be aware of the following when making your pull request:

## Description

Please include a summary of your proposed changes and which issue is being addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Issue Number

Internal Users - Please include the related internal tracking number (e.g. BG-000000).
External Users - Please link to any relevant github issues as necessary.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
-->

mohammadalfaiyazbitgo committed:
chore(sdk-core): refactor & split challenge verification method

Add method to sign bitgo challenge without verification. To be used
in contexts where the caller will verify challenges/proofs separately from
signing.

Ticket: BG-77776

zahin-mohammad reviewed (dismissed):
lgtm. Existing function was split up, so the existing tests still cover these changes.

mohammadalfaiyazbitgo committed:
fix: update vm2 version

Updating vm2 from 3.9.18 to adress vulnerability
https://github.com/advisories/GHSA-whpj-8f3w-67p5

Ticket: BG-77776

pranavjain97 commented on package.json:
```diff
@@ -69,7 +69,7 @@
     "parse-path": "^5.0.0",
     "parse-url": "8.1.0",
     "terser": "^5.14.2",
-    "vm2": "^3.9.17",
```
was this intentional?

mmcshinsky-bitgo replied:
Security audit update.

zahin-mohammad replied:
https://bitgo.slack.com/archives/C010AEXLLCR/p1684247091685949?thread_ts=1684246605.689579&cid=C010AEXLLCR

zahin-mohammad replied:
@mohammadalfaiyazbitgo shouldn't there be a package-lock change as well with this? Did you run `yarn`?

mohammadalfaiyazbitgo replied:
yeah forgot that, I'll update

mohammadalfaiyazbitgo committed:
chore: fix tss unit test

Also update yarn.lock

Ticket: BG-77776
